projects/packages/my-jetpack/_inc/components/product-cards-section/index.tsx

```diff
@@ -90,17 +92,48 @@ const DisplayItems: FC< DisplayItemsProps > = ( { slugs } ) => {
 
 interface ProductCardsSectionProps {
 	noticeMessage: ReactNode;
+	siteIsRegistered?: boolean;
 }
 
-const ProductCardsSection: FC< ProductCardsSectionProps > = ( { noticeMessage } ) => {
-	const { ownedProducts = [], unownedProducts = [] } =
+const ProductCardsSection: FC< ProductCardsSectionProps > = ( {
+	noticeMessage,
+	siteIsRegistered,
+} ) => {
+	const { ownedProducts: initialOwnedProducts = [], unownedProducts: initialUnownedProducts = [] } =
 		getMyJetpackWindowInitialState( 'lifecycleStats' );
+	const [ productsOwnership, setProductsOwnership ] = useValueStore( 'productsOwnership', {
```

Comment:
We can still use that structure if we want, I think it's just good to make the updates atomic (which they are now).

```suggestion
	const [ { ownedProducts, unownedProducts }, setProductsOwnership ] = useValueStore( 'productsOwnership', {
```